File: 2 CPP/Leetcode-3432-CountPartitionsWithEvenSumDifference.cpp
class Solution {
public:
    int countPartitions(vector<int>& nums) {
        int lsum = nums[0];
        int rsum = accumulate(nums.begin(),nums.end(),0)-lsum;
        int ans = 0;
        for(int i=1; i<nums.size();++i){
            int cn = nums[i];
            lsum+=cn;
            rsum-=cn;
            ans+=abs(lsum-rsum)%2==0;
        }
        return ans;
    }
};

File: 1 Python/Leetcode-3432-CountPartitionsWithEvenSumDifference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        lsum = nums[0]
        rsum = sum(nums)-lsum
        output = 0
        for n in nums[1:]:
            lsum+=n
            rsum-=n
            if (lsum-rsum)%2==0:
                output+=1
        return output

class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        lsum = 0
        rsum = sum(nums)
        output = 0
        for n in nums[1:]:
            if (lsum-rsum)%2==0:
                output+=1
            lsum+=n
            rsum-=n
        return output
